include/uart.h

```diff
@@ -245,16 +436,164 @@ struct uart_driver_api {
+				    void *user_data)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->callback_set(dev, callback, user_data);
+}
+
+/**
+ * @brief Send given number of bytes from buffer through UART.
+ *
+ * Function returns immediately and event handler,
+ * set using @ref uart_set_callback, is called after transfer is finished.
+ *
+ * @param dev     UART device structure.
+ * @param buf     Pointer to transmit buffer.
+ * @param len     Length of transmit buffer.
+ * @param timeout Timeout in milliseconds.
+ *
+ * @retval -EBUSY There is already an ongoing transfer.
+ * @retval 0	  If successful, negative errno code otherwise.
+ */
+static inline int uart_tx(struct device *dev,
+			  u8_t *buf,
+			  size_t len,
+			  u32_t timeout)
+
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->tx(dev, buf, len, timeout);
+}
+
+/**
+ * @brief Abort current TX transmission.
+ *
+ * UART_TX_DONE event will be generated with amount of data sent.
+ *
+ * @param dev UART device structure.
+ *
+ * @retval -EFAULT There is no active transmission.
+ * @retval 0	   If successful, negative errno code otherwise.
+ */
+static inline int uart_tx_abort(struct device *dev)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->tx_abort(dev);
+}
+
+/**
+ * @brief Start receiving data through UART.
+ *
+ * Function sets given buffer as first buffer for receiving and returns
+ * immediately. After that event handler, set using @ref uart_set_callback,
+ * is called with UART_RX_RDY or UART_RX_BUF_REQUEST events.
+ *
+ * @param dev UART device structure.
+ * @param buf Pointer to receive buffer.
+ * @param len Buffer length.
+ *
+ * @retval -EBUSY RX already in progress.
+ * @retval 0	  If successful, negative errno code otherwise.
+ *
+ */
+static inline int uart_rx_enable(struct device *dev, u8_t *buf, size_t len)
+{
+	const struct uart_driver_api *api =
+				(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_enable(dev, buf, len);
+}
+
+/**
+ * @brief Provide receive buffer in response to UART_RX_BUF_REQUEST event.
+ *
+ * Provide pointer to RX buffer, which will be used when current buffer is
+ * filled.
+ *
+ * @note Providing buffer that is already in usage by driver leads to
+ *       undefined behavior. Buffer can be reused when it has been released
+ *       by driver.
+ *
+ * @param dev UART device structure.
+ * @param buf Pointer to receive buffer.
+ * @param len Buffer length.
+ *
+ * @retval -EBUSY Next buffer already set.
+ * @retval 0	  If successful, negative errno code otherwise.
+ *
+ */
+static inline int uart_rx_buf_rsp(struct device *dev, u8_t *buf, size_t len)
+{
+	const struct uart_driver_api *api =
+				(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_buf_rsp(dev, buf, len);
+}
+
+/**
+ * @brief Set RX timeout function.
```

Comment:
why not combine this function with rx_enable? If it's separated then there is always a problem when can it be called.